public/js/routing/router.js

```diff
@@ -15,10 +15,12 @@
 
 /* eslint-env browser */
 
-export default class Router {
-  constructor(window, shell, container) {
+import EventTarget from 'event-target-shim/dist/event-target-shim.min.js';
+
+export default class Router extends EventTarget {
```

Comment:
The inheritance over composition is an interesting choice, even if it is following the standard…